### PR TITLE
DRY Code and MaterialIntegerBox

### PR DESCRIPTION
**Pull request by paulux84**
DRY Code for MaterialTextArea.java and MaterialTextBox.java that now extends MaterialValueBox.java.

So i have created MaterialIntegerBox.java with EditorSupport and also add Editor Support to MaterialTextArea.java extending MaterialValueBox.java

### CODE REVIEW

**Comment by BenDol** [comment]
Great! This was something I wanted to do but ran out of time, will review it now.

File: gwt-material/src/main/java/gwt/material/design/client/ui/MaterialIntegerBox.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+* @author kevzlou7979
+* @author Ben Dol
+*/
+//@formatter:on
+public class MaterialIntegerBox extends MaterialValueBox<Integer> {
+
+
+    public MaterialIntegerBox() {
+        super(new IntegerBox());
+        setType(InputType.TEXT);
```
Comment (paulux84) [inline gwt-material/src/main/java/gwt/material/design/client/ui/MaterialIntegerBox.java]: This type must be NUMBER

**Comment by paulux84** [comment]
@gilberto-torrezan it's a <input type="number"> yet

**Comment by BenDol** [comment]
Thanks for the PR @paulux84, is this now ready to be merged?

**Comment by dlaunen** [comment]
I would suggest that we might want to abstract this a little bit more.  For example: MaterialIntegerBox could extend MaterialNumericBox<T extends Number> which extends MaterialValueBox.  MaterialNumericBox can then implement common functionality with numbers, such as formatting, validation etc.

**Comment by kevzlou7979** [comment]
Yes I agree

**Comment by BenDol** [comment]
@paulux84 if you would like to abstract this as mentioned above let us know, otherwise I can merge this PR and someone else can take care of it. Thanks for your contribution so far.

**Comment by paulux84** [comment]
@BenDol i've already do it past week, but not pushed it into repository.
Wait a little longer....I'm making some other changes.
I will warn you when ready for merge!

**Comment by BenDol** [comment]
@paulux84 Okay, we are planning for the 1.4.1 release and would like to have this PR merged before then. Let me know how long this will be. Thanks!

**Comment by paulux84** [comment]
@BenDol sorry for being late. Today i do the commit

**Comment by paulux84** [comment]
@BenDol @kevzlou7979 ready :+1: :smile:

**Comment by BenDol** [comment]
Thanks @paulux84 great job :+1:

File: gwt-material/src/main/java/gwt/material/design/client/ui/MaterialTextArea.java [hunk 2]
```diff
@@ -48,203 +34,34 @@
...
-        lblName.setText(placeholder);
-    }
 
-    @Override
-    public HandlerRegistration addKeyDownHandler(KeyDownHandler handler) {
-        return addDomHandler(handler, KeyDownEvent.getType());
-    }
+    //OLD MaterialTextArea method
+//    @Override
+//    protected void onLoad() {
```
Comment (BenDol) [inline gwt-material/src/main/java/gwt/material/design/client/ui/MaterialTextArea.java]: Can this be removed?

File: gwt-material/src/main/java/gwt/material/design/client/ui/MaterialValueBox.java [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+        setStyleName("input-field");
+        add(icon);
+        initValueBox(tValueBox);
+        add(label);
+        label.add(lblName);
+        lblError.setVisible(false);
+        add(lblError);
+    }
+
+//    public MaterialValueBox() {
```
Comment (BenDol) [inline gwt-material/src/main/java/gwt/material/design/client/ui/MaterialValueBox.java]: Can this be removed?

**Comment by BenDol** [comment]
We will merge this now and make a new PR if you have any other changes, thanks!

**Pull request merged**